cpp/src/io/avro/reader_impl.cu

```diff
@@ -464,19 +464,20 @@ table_with_metadata reader::impl::read(int skip_rows, int num_rows, cudaStream_t
 }
 
 // Forward to implementation
-reader::reader(std::string filepath,
+reader::reader(std::vector<std::string> const &filepaths,
                reader_options const &options,
                rmm::mr::device_memory_resource *mr)
-  : _impl(std::make_unique<impl>(datasource::create(filepath), options, mr))
+  : _impl(std::make_unique<impl>(datasource::create(filepaths[0]), options, mr))
```

Comment:
similar `CUDF_EXPECTS` should be present. 
`CUDF_EXPECTS(sources.size() == 1, "Only a single source is currently supported.");`